addons/website/static/src/snippets/s_image_gallery/options.js

```diff
@@ -360,10 +365,17 @@ options.registry.gallery = options.Class.extend({
                 });
                 $carousel.addClass('slide');
             } else {
-                this.trigger_up('activate_snippet', {
-                    $snippet: data.$image,
-                    ifInactiveOptions: true,
-                });
+                // Wait for snippet_option_update to be completed.
+                this.trigger_up('snippet_edition_request', {exec: () => {
+                    // Wait for images which might have been cloned.
+                    (this.lastModeUpdate || Promise.resolve()).then(() => {
```

Comment:
`Promise.resolve()).then(()`
Is that like a setTimeOut(0) ? Why is it needed ?

Reply:
This is just to avoid writing:
```js
const doStuff = () => {...};
if (this.lastModeUpdate) {
    this.lastModeUpdate.then(doStuff);
} else {
    doStuff();
}
```

(TL;DR: no event nor pending `setTimeout` is processed between `Promise.resolve().then(doStuff)` and `doStuff`.)

Regarding `setTimeout()`, it is not strictly equivalent because the `Promise` chain will be queued separately. 
I am not sure about the correct wording, but here is a short description of my understand. It is definitely not 100% accurate, but it should give a feel of what is going on:

When you use `setTimeout` it is like registering an event listener for a single use, its trigger condition being "reached system time >= current current system + delta ms". With the single threaded nature of Javascript, when events happen, they are only queued, they are not executed right away.  The next event is processed only after the previous event's processing is completed.

When using `Promise`, they are queued distinctly from events. Before fetching the next event, the Promises are executed, they are requeued if they await something that is not resolved, and if all remaining Promises are awaiting it then proceeds to the next queued event.

Here is another explanation with a few diagrams: <https://medium.com/@idineshgarg/let-us-consider-an-example-a58bb1c11f55> which might be more clear (did not read it yet)